packages/cozy-client/src/models/file.js

```diff
@@ -152,3 +153,18 @@ export const isSharingShorcut = file => Boolean(getSharingShortcutStatus(file))
  */
 export const isSharingShorcutNew = file =>
   getSharingShortcutStatus(file) === 'new'
+
+/**
+ * Save the file with the given qualification
+ *
+ * @param {object} client - The CozyClient instance
+ * @param {object} file - The file to qualify
+ * @param {object} qualification - The file qualification
+ * @returns {object} - The saved file
+ */
+export const saveFileQualification = async (client, file, qualification) => {
+  const qualifiedFile = setQualification(file, qualification)
+  return client
+    .collection('io.cozy.files')
+    .updateMetadataAttribute(file._id, qualifiedFile.metadata)
+}
```

Comment:
What happens if : 
```js 

const file = client.get('io.cozy.files', 1)
const qualification = new Qualification(label)
Document.setQualification(file, qualification)
client.save(file) 
```

Does the new qualification will be set? Should we handle the case in FileCollection ? (I think yes).

If we can handle the case in FileCollection, I'm wondering if we need to create this new method

Reply:
Yes the qualification will be set and saved. What do you mean by handling the case in FileCollection?

Reply:
If this is works by using FileCollection.save() what is the goal to use `saveFileQualification` ?

Reply:
Just a helper to avoid having to manually call `Document.setQualification(file, qualification)`. But if you think it's useless or introduce confusion, I can remove

Reply:
Not sure it works with save() in fact. 

By doing save() we call : 
https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L352-L354

and then updateAttributes https://github.com/cozy/cozy-client/blob/96f8b97d918ef950439dfd3a90e8d50d9d2b6856/packages/cozy-stack-client/src/FileCollection.js#L710-L713 but this method seems to not work for metadata since metadata is "protected". 

So I'm wondering if we should not be able to update metadata from save() directly and remove this helper.

Reply:
After discussion with @Crash--, it's complicated to use  `save` for this, because we cannot always know what changed in the document (as the doc might not be in the redux store), and it's not easy to change the save API because of the DSL. So we keep this helper for now

Reply:
>  because we cannot always know what changed in the document 

Why do we need to know that ?